### PR TITLE
purge Chef client, in addition to Chef node

### PR DESCRIPTION
**Pull request by lidl**
Signed-off-by: Kurt Lidl <kurt.lidl@oracle.com>

### CODE REVIEW

File: lib/chef/knife/oci_server_delete.rb [hunk 1]
```diff
@@ -82,6 +83,11 @@ def delete_chef_node(node)
         ui.msg "Deleted Chef node '#{node.name}'"
       end
 
+      def delete_chef_client(client_name)
```
Comment (briangustafson) [inline lib/chef/knife/oci_server_delete.rb]: Should include a message as in delete_chef_node (unless something is being printed by the all already being made?).

File: lib/chef/knife/oci_server_delete.rb [hunk 2]
```diff
@@ -82,6 +83,11 @@ def delete_chef_node(node)
...
+        object = Chef::ApiClient.load(client_name)
+        object.destroy unless object.validator
```
Comment (briangustafson) [inline lib/chef/knife/oci_server_delete.rb]: What happens if the client or node with this name is not found? Is it handled in a graceful and easy to diagnose way?

Comment (lidl) [reply lib/chef/knife/oci_server_delete.rb]: If the node isn't found, you get this message:

Instance name: [instance name]
ERROR: The object you are looking for could not be found
Response: node '[node-name]' not found

And nothing is deleted.

If the node name is found, and the client name isn't found, you get a message:

ERROR: The object you are looking for could not be found
Response: Cannot load client [client-name]

But that diagnostic message is printed after the instance is deleted, and after
the Chef node is deleted.

File: lib/chef/knife/oci_server_delete.rb (hunk 1 above)
Comment (lidl) [reply lib/chef/knife/oci_server_delete.rb]: The most recent update to this pull request does issue a message when
deleting the 'client' from Chef.  (There was already a message for the 'node'
deletion.)

**Pull request merged**